api-docs/interface/server-socket.md

```diff
@@ -8,31 +8,56 @@ Connect to the websocket at `/connect?gameId={game ID}&playerId={player ID}`
 
 ### Data Structs:
 
-<!-- This should move somewhere else (the game object?) -->
-`connections` (`Map<string, Map<string, WebSocket>>`)
 
 - Every message is a JSON object with type as a string, and potentially other fields depending on the type.
 
 - When a client connects to the server and the game has already started, the server immediately sends a question or endQuestion (or lobby if client is the host) message to catch them up to the current state of the game. Before then, the client just displays “Waiting for the host to start the game” or something. This way the same logic handles both “client joined and is waiting for game to start” and “client joined late and is waiting for the server to respond to them.”
 
-## 
+# Types
 
 These messages are the same for players and the host unless specified.
 
 Possible types (all sent from server to client unless otherwise specified):
-end
 
-Other fields: reason (string)
+
+## `end`
+
 Whoever sends this message is closing the connection for the specified reason. E.g. the server could send {“type”: “end”, “reason”: “That game does not exist”}
-startQuestion
 
-Other fields: questionText, answerText (same as in JSON), endTimestamp: numeric UNIX timestamp (milliseconds) when the timer to answer this question will out, score (number): player’s current score
+Fields: 
+- reason (string)
+
+## `startQuestion`
+
 Sent by server whenever a new question is available
-endQuestion
-Fields: correct (boolean), correctAnswers (numeric array), score (number) = the player’s current score, scoreChange (number) = how much their score increased due to this question, 
-Host only: leaderboard: sorted array of [{name: string, score: number }, …]
+
+Fields: 
+- `questionText`: The text for the question
+- `answerTexts`: The answer texts (same as in JSON)
+- `time` (number): The number of *miliseconds* that are left in the question (where 0 is the end of the question)
+- `score` (number): player’s current score, the host will recieve garbage
+- `username`: the name of the player. 
+
+## `endQuestion`
```

Comment:
Should include username too

Reply:
And questionText/answerTexts. Also clarify what goes to which clients. Do players get the leaderboard? I don't think so

Reply:
And questionIndex